python/lsst/ctrl/mpexec/mpGraphExecutor.py

```diff
@@ -227,24 +231,24 @@ def __init__(self, numProc, timeout, quantumExecutor, *, failFast=False, executi
 
     def execute(self, graph, butler):
         # Docstring inherited from QuantumGraphExecutor.execute
-        quantaIter = self._fixupQuanta(graph.traverse())
+        graph = self._fixupQuanta(graph)
         if self.numProc > 1:
-            self._executeQuantaMP(quantaIter, butler)
+            self._executeQuantaMP(graph, butler)
         else:
-            self._executeQuantaInProcess(quantaIter, butler)
+            self._executeQuantaInProcess(graph, butler)
 
-    def _fixupQuanta(self, quantaIter):
+    def _fixupQuanta(self, graph: QuantumGraph):
         """Call fixup code to modify execution graph.
 
         Parameters
         ----------
-        quantaIter : iterable of `~lsst.pipe.base.QuantumIterData`
-            Quanta as originated from a quantum graph.
+        graph : `QuantumGraph`
+            `QuantumGraph` to modify
 
         Returns
         -------
-        quantaIter : iterable of `~lsst.pipe.base.QuantumIterData`
-            Possibly updated set of quanta, properly ordered for execution.
+        graph : `QuantumGraph`
+            Modified `QuantumGraph`, properly ordered for execution.
```

Comment:
I think new QuantumGRaph cannot be non-ordered, may be drop this "properly ordered..."